nestor_api/lib/io.py

```diff
@@ -34,46 +34,46 @@ def create_temporary_directory(directory_prefix=""):
     return tmp_directory_path
 
 
-def ensure_dir(directory_path):
+def ensure_dir(directory_path: str) -> None:
     """Ensures that a directory exists, else creates it"""
     Path(directory_path).mkdir(parents=True, exist_ok=True)
 
 
-def execute(command: str, cwd=None):
+def execute(command: str, cwd: str = None) -> str:
     """Executes a command and returns the stdout from it"""
     result = subprocess.run(
         command.split(), stdout=subprocess.PIPE, stderr=subprocess.DEVNULL, check=True, cwd=cwd,
     )
     return result.stdout.decode("utf-8").rstrip()
 
 
-def exists(file_path):
+def exists(file_path: str) -> bool:
     """Checks if a file exists"""
     return Path(file_path).exists()
 
 
-def from_yaml(file_path):
+def from_yaml(file_path: str) -> dict:
```

Comment:
What do you think of improving this function's name while you're at it ? (I find the `io.from_yaml()` a bit weird personally).
Maybe something similar to "parse_yaml" or "read_yaml" (and then the `to_yaml` you added becomes "write_yaml").